Services.Test/CustomDeviceModelsTest.cs

```diff
@@ -0,0 +1,342 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using Microsoft.Azure.IoTSolutions.DeviceSimulation.Services;
+using Microsoft.Azure.IoTSolutions.DeviceSimulation.Services.Diagnostics;
+using Microsoft.Azure.IoTSolutions.DeviceSimulation.Services.Exceptions;
+using Microsoft.Azure.IoTSolutions.DeviceSimulation.Services.Models;
+using Microsoft.Azure.IoTSolutions.DeviceSimulation.Services.StorageAdapter;
+using Moq;
+using Newtonsoft.Json;
+using Services.Test.helpers;
+using System;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Services.Test
+{
+    public class CustomDeviceModelsTest
+    {
+        private const string STORAGE_COLLECTION = "deviceModels";
+
+        private readonly ITestOutputHelper log;
+        private readonly Mock<IStorageAdapterClient> storage;
+        private readonly Mock<ILogger> logger;
+        private readonly CustomDeviceModels target;
+
+        public CustomDeviceModelsTest(ITestOutputHelper log)
+        {
+            this.log = log;
+
+            this.storage = new Mock<IStorageAdapterClient>();
+            this.logger = new Mock<ILogger>();
+
+            this.target = new CustomDeviceModels(
+                this.storage.Object,
+                this.logger.Object);
+        }
+
+        [Fact, Trait(Constants.TYPE, Constants.UNIT_TEST)]
+        public void InitialListIsEmpty()
+        {
+            // Arrange
+            this.ThereAreNoCustomDeviceModelsInStorage();
+
+            // Act
+            var result = this.target.GetListAsync().Result;
+
+            // Assert
+            Assert.Empty(result);
+        }
+
+        [Fact, Trait(Constants.TYPE, Constants.UNIT_TEST)]
+        public void ItCreatesCustomDeviceModelsWithProvidedIdInStorage()
+        {
+            // Arrange
+            var deviceModel = new DeviceModel { Id = "id", ETag = "Etag_1" };
+            this.UpdateDeviceModelInStorage(deviceModel);
+
+            // Act
+            DeviceModel result = this.target.InsertAsync(deviceModel, false).Result;
+
+            // Assert
+            Assert.NotNull(result);
+            Assert.Equal(deviceModel.Id, result.Id);
+            Assert.Equal(deviceModel.ETag, result.ETag);
+        }
+
+        [Fact, Trait(Constants.TYPE, Constants.UNIT_TEST)]
+        public void ItCreatesCustomDeviceModelsWithGuidInStorage()
```

Comment:
nit: rename test, instead of Guid => ETag

Reply:
Done.